analyses/snv-callers/scripts/01-setup_db.py

```diff
@@ -0,0 +1,282 @@
+    ('DOMAINS', 'TEXT'),
+    ('AF', 'TEXT'),
+    ('AFR_AF', 'TEXT'),
+    ('AMR_AF', 'TEXT'),
+    ('ASN_AF', 'TEXT'),
+    ('EAS_AF', 'TEXT'),
+    ('EUR_AF', 'TEXT'),
+    ('SAS_AF', 'TEXT'),
+    ('AA_AF', 'TEXT'),
+    ('EA_AF', 'TEXT'),
+    ('CLIN_SIG', 'TEXT'),
+    ('SOMATIC', 'TEXT'),
+    ('PUBMED', 'TEXT'),
+    ('MOTIF_NAME', 'TEXT'),
+    ('MOTIF_POS', 'INTEGER'),
+    ('HIGH_INF_POS', 'TEXT'),
+    ('MOTIF_SCORE_CHANGE', 'REAL'),
+    ('IMPACT', 'TEXT'),
+    ('PICK', 'TEXT'),
+    ('VARIANT_CLASS', 'TEXT'),
+    ('TSL', 'TEXT'),
+    ('HGVS_OFFSET', 'TEXT'),
+    ('PHENO', 'TEXT'),
+    ('MINIMISED', 'TEXT'),
+    ('ExAC_AF', 'TEXT'),
+    ('ExAC_AF_AFR', 'TEXT'),
+    ('ExAC_AF_AMR', 'TEXT'),
+    ('ExAC_AF_EAS', 'TEXT'),
+    ('ExAC_AF_FIN', 'TEXT'),
+    ('ExAC_AF_NFE', 'TEXT'),
+    ('ExAC_AF_OTH', 'TEXT'),
+    ('ExAC_AF_SAS', 'TEXT'),
+    ('GENE_PHENO', 'TEXT'),
+    ('FILTER', 'TEXT'),
+    ('flanking_bps', 'TEXT'),
+    ('vcf_id', 'TEXT'),
+    ('vcf_qual', 'REAL'),
+    ('ExAC_AF_Adj', 'TEXT'),
+    ('ExAC_AC_AN_Adj', 'TEXT'),
+    ('ExAC_AC_AN', 'TEXT'),
+    ('ExAC_AC_AN_AFR', 'TEXT'),
+    ('ExAC_AC_AN_AMR', 'TEXT'),
+    ('ExAC_AC_AN_EAS', 'TEXT'),
+    ('ExAC_AC_AN_FIN', 'TEXT'),
+    ('ExAC_AC_AN_NFE', 'TEXT'),
+    ('ExAC_AC_AN_OTH', 'TEXT'),
+    ('ExAC_AC_AN_SAS', 'TEXT'),
+    ('ExAC_FILTER', 'TEXT'),
+    ('gnomAD_AF', 'TEXT'),
+    ('gnomAD_AFR_AF', 'TEXT'),
+    ('gnomAD_AMR_AF', 'TEXT'),
+    ('gnomAD_ASJ_AF', 'TEXT'),
+    ('gnomAD_EAS_AF', 'TEXT'),
+    ('gnomAD_FIN_AF', 'TEXT'),
+    ('gnomAD_NFE_AF', 'TEXT'),
+    ('gnomAD_OTH_AF', 'TEXT'),
+    ('gnomAD_SAS_AF', 'TEXT'),
+    ('vcf_pos', 'INTEGER'),
+    ('VAF', 'REAL')
+]
+
+needed_cols = [
+    'Hugo_Symbol',
+    'Entrez_Gene_Id',
+    'Chromosome',
+    'Start_Position',
+    'End_Position',
+    'Variant_Classification',
+    'Variant_Type',
+    'Reference_Allele',
+    'Tumor_Seq_Allele1',
+    'Tumor_Seq_Allele2',
+    'dbSNP_RS',
+    'Tumor_Sample_Barcode',
+    'Transcript_ID',
+    't_depth',
+    't_ref_count',
+    't_alt_count',
+    'n_depth',
+    'n_ref_count',
+    'n_alt_count',
+    'Allele',
+    'flanking_bps',
+    'VAF'
+]
+
+needed_types = [col for col in maf_types if col[0] in needed_cols]
+
+# indexes to create. lists to handle multiple column indexes.
+maf_indexes = [
+    ('SNV', ['Chromosome', 'Start_Position', 'Reference_Allele', 'Allele']),
+    ('sample', ['Tumor_Sample_Barcode'])
+]
+
+# translate sqltypes to dtypes
+dtypes = {
+    'TEXT': 'object',
+    'INTEGER': 'Int64',
+    'REAL': 'float64'
+}
+maf_dtypes = {col: dtypes[sql_type] for col, sql_type in maf_types}
+
+# connect to/create the db
+con = sqlite3.connect(args.db_file)
+
+chunksize = 1e5
+
+# Create caller list
+callers = [
+    ('strelka', args.strelka_file),
+    ('mutect', args.mutect_file),
+    ('lancet', args.lancet_file),
+    ('vardict', args.vardict_file)
+]
+
+for table_name, maf_file in callers:
+    if not maf_file:
```

Comment:
I don't understand this part? Can you explain it in the context of this loop?

Reply:
To be more specific, I don't understand lines 244- 249? Why does it look like lancet and strelka being treated differently than the other two?

Reply:
There's a comment there! I want to have some full tables, but some with only the `needed_cols`. Since we will later plan to pull from strelka, that one is complete. But as a backup, I picked lancet.

Reply:
The specific line there (`if not maf_file`) skips the rest of the loop if there was no specified maf file for that caller.